command/agent/config_parse.go

```diff
@@ -437,6 +437,12 @@ func parseClient(result **ClientConfig, list *ast.ObjectList) error {
 				return err
 			}
 		}
+
+		for k := range config.Meta {
+			if !helper.IsValidInterpVariable(k) {
```

Comment:
Config parsing should just parse; validating the parsed configuration should be done after.

I would move the validation here: https://github.com/hashicorp/nomad/blob/64f591875db7fead5c88e0b4d0df24338232d069/command/agent/command.go#L281

If you want to go big I would move: https://github.com/hashicorp/nomad/blob/64f591875db7fead5c88e0b4d0df24338232d069/command/agent/command.go#L287-L344 and https://github.com/hashicorp/nomad/blob/64f591875db7fead5c88e0b4d0df24338232d069/command/agent/command.go#L267-L274

To a validate function called there

Reply:
@dadgar : 267:274 (`TLSConfig.SetChecksum`) isn't "strong" validation (it doesn't return `nil`). Should it?

Reply:
@dadgar: I refactored as you suggested. It was helpful; among other things, it put the validation in one place (as opposed to before, where I was validating read-config separately from arg-config). Please re-review at your leisure.